### ISSUE TITLE
Map changing on PlayerWon and PlayerLeft even though not all players have completed map

### ISSUE THREAD

**Issue by cocomausi** [a801-luadev/parkour#10]
There is an issue where occasionally, when a player completes a map or leaves the room, map change is triggered even though not everyone has completed the map.

https://github.com/a801-luadev/parkour/blob/master/modes/parkour/game.lua#L51
https://github.com/a801-luadev/parkour/blob/master/modes/parkour/game.lua#L73

**Comment by tocuto** [comment]
Thanks for taking a look at the code! However, this won't ever happen. I'm using victory_count to know the length of the victory table, counting only players in the room.

As you see, the victory table is only reset on eventNewGame (a round starts) and the only case of a player being removed from there is when a mod types `!kill player`, which is not pretty common. I might also add a fix for that tho.

Here: https://github.com/a801-luadev/parkour/blob/deb491a6cdcd0726f66b2dbc491fcdf4c8f465c4/modes/parkour/game.lua#L49-L50 when the player leaves I make a check. If the player has entered the hole, it needs to subtract 1 to victory_count. And here: https://github.com/a801-luadev/parkour/blob/deb491a6cdcd0726f66b2dbc491fcdf4c8f465c4/modes/parkour/game.lua#L29-L30 when the player re-joins I make another check, if they have already entered the hole, add 1 to victory_count. Note that player_count is always increased or decreased.

So, in summary: When a player leaves the room it subtracts 1 to player_count, and if they've won the map, it subtracts 1 to victory_count too. When they join the room, it adds 1 to player_count, and if they've won the map (the map didn't end yet) it adds 1 to victory_count too.

**Comment by cocomausi** [comment]
Hi, I read it incorrectly. I'm not sure of the exact cause, but I observed the map changing early when one player had not yet completed the room in these two scenarios, in rooms where there was no mod. I will try to get a screen capture.

**Comment by tocuto** [comment]
It might be some parkour-mod who typed !kill as I've told you before. I'd be glad if you could get a gif!

**Comment by cocomausi** [comment]
Recording of the issue occurring with PlayerWon - the timer goes from 11:57 to 0:19. In this particular recording, there are only 3 users in the room, but this happened before with 4 users in the room.

![unknown_2](https://user-images.githubusercontent.com/35877813/76173894-ffdf9000-6179-11ea-9d71-b1acf9bc6ef5.gif)

**Comment by tocuto** [comment]
Thank you for reporting. It will take me quite long to find out what's happening there.
